bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/MagentaTVHandlerFactory.java

```diff
@@ -0,0 +1,374 @@
+        int i = 0;
+        for (String key : deviceList.keySet()) {
+            synchronized (deviceList) {
+                dev = deviceList.get(key);
+                Validate.notNull(dev);
+                if (dev != null) {
+                    logger.trace("Devies[{0}]: deviceId={1}, UDN={2}, ipAddress={3}, macAddress={4}", i++, dev.deviceId,
+                            dev.udn, dev.ipAddress, dev.mac);
+                    if (dev.udn.equalsIgnoreCase(uniqueId) || dev.ipAddress.equalsIgnoreCase(uniqueId)
+                            || dev.deviceId.equalsIgnoreCase(uniqueId) || dev.mac.equalsIgnoreCase(uniqueId)) {
+                        return dev;
+                    }
+                }
+            }
+        }
+        logger.debug("Device with id {0} was not found in table ({1} entries", uniqueId, deviceList.size());
+        return null;
+    }
+
+    /**
+     * returned the discovered properties
+     *
+     * @param udn Unique ID from UPnP discovery
+     * @return property map with discovered properties
+     */
+    @SuppressWarnings({ "null", "unused" })
+    public @Nullable Map<String, Object> getDiscoveredProperties(String udn) {
+        MagentaTVDevice dev = deviceList.get(udn.toUpperCase());
+        if (dev != null) {
+            return dev.properties;
+        }
+        if (deviceList.size() > 0) {
+            logger.debug("getDiscoveredProperties(): Unknown UDN: {0}", udn);
+        }
+        return null;
+    }
+
+    public void setNotifyServletStatus(boolean newStatus) {
+        logger.debug("NotifyServlet started");
+        servletInitialized = newStatus;
+    }
+
+    public boolean getNotifyServletStatus() {
+        return servletInitialized;
+    }
+
+    /**
+     * We received the pairing resuled (by the Norify servlet)
+     *
+     * @param notifyDeviceId The unique device id pairing was initiated for
+     * @param pairingCode Pairng code computed by the receiver
+     * @return true: thing handler was called, false: failed, e.g. unknown device
+     */
+    @SuppressWarnings("null")
+    public boolean notifyPairingResult(String notifyDeviceId, String ipAddress, String pairingCode) {
+        try {
+            logger.trace("PairingResult: Check {0} devices for id {1}, ipAddress {2}", deviceList.size(),
+                    notifyDeviceId, ipAddress);
+            MagentaTVDevice dev = lookupDevice(ipAddress);
+            if ((dev != null) && (dev.thingHandler != null)) {
+                if ((dev.deviceId == null) || dev.deviceId.isEmpty()) {
+                    logger.trace("deviceId {0} assigned for ipAddress {1}", notifyDeviceId, ipAddress);
+                    dev.deviceId = notifyDeviceId;
+                }
+                dev.thingHandler.onPairingResult(pairingCode);
+                return true;
+            }
+
+            logger.fatal("Received pairingCode {0} for unregistered device {1}!", pairingCode, ipAddress);
```

Comment:
change to logger .debug

Reply:
that could be a security issue or indicate a miss-configuration (e.g. multiple apps in the network using the same MR)

Reply:
Level WARN instead